packages/core/database/src/index.ts

```diff
@@ -82,9 +85,11 @@ class Database {
 
     this.metadata = createMetadata([]);
 
-    this.connection = createConnection(this.config.connection, {
-      pool: { afterCreate: afterCreate(this) },
-    });
+    // for object connections, create Knex connection synchronously
```

Comment:
could you specify a bit more why we would need a function? 🙏 
A note explaining that we support async conections by deffining the connection as a function

Reply:
If you check the linked issue you can find some use cases directly from users, but it's necessary for some database authentication methods

Reply:
ah , I meant on the code itself, so its easy to understand why we do this 🙈